src/Runtime/FileHandlerLocator.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
+
+namespace Bref\Runtime;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * Default Bref behavior to resolve "Lambda function handlers".
+ *
+ * With this behavior, Bref expects that handlers are PHP file names.
+ * Whenever a Lambda function executes, this class will `require` that PHP file
+ * and return what the file returns.
+ *
+ * @see \Bref\Bref::setContainer()
+ *
+ * @internal
+ */
+class FileHandlerLocator implements ContainerInterface
+{
+    /** @var string */
+    private $directory;
+
+    public function __construct(string $directory = null)
+    {
+        $this->directory = $directory ?: $_SERVER['LAMBDA_TASK_ROOT'];
+    }
+
+    public function get($id)
+    {
+        $handlerFile = $this->directory . '/' . $id;
+        if (! is_file($handlerFile)) {
+            throw new HandlerNotFound("Handler `$handlerFile` doesn't exist");
+        }
+
+        /** @noinspection PhpIncludeInspection */
+        $handler = require $handlerFile;
+
+        if (! (is_object($handler) || is_array($handler))) {
```

Comment:
I may be wrong but I don't see this logic covered in the test?

Reply:
I see the test but it's not covering all the logical paths here.

Reply:
These lines were moved from [the original bootstrap file from master](https://github.com/brefphp/bref/blob/master/runtime/layers/function/bootstrap.php#L26-L30).

Then I added [that test](https://github.com/brefphp/bref/pull/694/files#diff-02500a3d32ab6aaea871128f38c66c60R22-R29) and realized that the code on master doesn't actually work (`if (! $handler) {` doesn't work because the `require` call actually return `1` instead of `null`). That's why I changed the logic to `is_object() || is_array()`.

So, that piece of code isn't extensively tested, it just covers the basic. But good point I'll keep this thread open to remember to add more tests (allow object | array).

Reply:
The reason I think a test would be nice is I spent 10 minutes on 3v4l.org figuring out what that line meant, i think it's if it isn't an object or an array which confused me further because I thought it could also be a callable (I think I'm wrong there). Not a big deal tho.

Reply:
Yes you're right, but AFAIR here is what a callable could be:

- object implementing `__invoke()`
- closure, which is an object
- array (object + method name)
- string (function name) -> I'm fine not supporting this to avoid any weird case, I don't want for example to check that the string contains the name of a function that exists

And of course, a handler can be an object implementing `Handler`.

Hence why I simplified this to object | array.

Reply:
It makes sense, thanks for explaining I appreciate your time!